app/src/main/kotlin/com/mapzen/erasermap/controller/MainActivity.kt

```diff
@@ -711,11 +711,15 @@ class MainActivity : AppCompatActivity(), MainViewController,
     }
 
     override fun addSearchResultsToMap(features: List<Feature>?, activeIndex: Int) {
-        if (features == null) {
+        if (features == null || features.size == 0) {
```

Comment:
In the spirit of this refactor what do you think about moving both of these checks into the presenter so they can also be covered with unit tests? If `features` is either null or size 0 this method would never get called. Then the `features` param could also be declared non-null.

Also it seems like there is an opportunity to clean up the relationship between this method `MainActivity.addSearchResultsToMap(features: List<Feature>?, activeIndex: Int)` and `MainActivity.showSearchResults(features: List<Feature>?)`.

Both get called from the `MainPresenter` at different times. In addition`MainActivity.showSearchResults(features: List<Feature>?)` directly invokes `MainActivity.addSearchResultsToMap(features: List<Feature>?, activeIndex: Int)`.

Some questions that come to mind:
* What is the difference between these two methods?
* Why is one called over the other in `MainPresenter`?
* Why does one call the other rather than the presenter calling both in sequence?
* Can they be combined into a single method?
* Can they be renamed/documented so the purpose is more clear?

We don't have to solve this issue right now but I am interested to get your thoughts. It could be the subject of a good follow-up pull request to this one.